docs/proposals/25-private-cloud-profiles.md

```diff
@@ -0,0 +1,299 @@
+          class: premium
+          usable: true
+        - name: gp3
+          class: standard
+          usable: true
+      regions:
+        - name: europe-central-1
+          zones:
+            - name: europe-central-1a
+            - name: europe-central-1b
+            - name: europe-central-1c
+        - name: europe-special
+          zones:
+            - name: europe-custom-1a
+            - name: europe-custom-1b
+            - name: europe-custom-1c
+```
+
+The rendering is done by a new custom controller registered to the `gardener-controller-manager`. It a merge conflict is to appear during the rendering process, it should be noted somewhere in the status and the parent cloud profile should be preferred.
+
+### Custom rbac verb
+
+To prevent users from entering arbitrary values in the `kubernetes` and `machineImages` fields, two custom rbac verbs may be introduced. It cloud then be checked if the user that is creating or updating the `kubernetes` or `machineImages` field is authorized to do so.
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: privatecloudprofile-modify-special-fields
+rules:
+  - apiGroups: [""]
+    resources: ["privatecloudprofile"]
+    verbs: ["modify-spec-kubernetes", "modify-spec-machineImages"]
+```
+
+### Automated removal of outdated Kubernetes versions and machine image versions
+
+Expired Kubernetes versions in cloud profiles are currently being removed manually. This is not an issue since there is a very limited amount of cloud profiles. However, it any project administrator is allowed to create private cloud profile which can then have its Kubernetes versions expiration dates extended, it cloud become a very cumbersome process to manually remove all the expired Kubernetes versions. It is therefore proposed to implement a custom controller managed my `gardener-controller-manager`, that reconciles private cloud profiles (and possibly even regular cloud profiles) and removes any Kubernetes versions that are past their expiration date. The same could be done for machine images versions.
+
+However, a problem arises as the kubernetes/machine image versions are referenced in the `providerConfig` and would therefore need to be removed manually anyways.
+
+### Cross project sharing
+
+A use case could be defined where a private cloud profile might want to shared across multiple projects and not just be used within the project it was created id. However this use case seems to be very slim so this functionality will not be implemented as of now. Although, when taking a broader view with Private Seeds and Cloud in Country, this feature is going to be necessary at some point. Still, is is not planned to be implemented as of this GEP.
+
+### Multi-inheritance
+
+Theoretically, a private cloud profile could inherit from a cloud profile that already inherits from a cloud profile. However, this should probably not be allowed since it presents some major challenges.
```

Comment:
Hmm... this is not multi-inheritance (btw, typo, hyphenated compounds in headline titles must be capitalised unless they are articles and the like). Multi-inheritance would be to inherit from 2 different parents.

What you mean is multi-level inheritance.

Reply:
Indeed, renamed in https://github.com/gardener/gardener/pull/9142/commits/3d6d6082d8cd253a0de5b4c174f50c911d56bf9e